debezium-core/src/main/java/io/debezium/converters/CloudEventsConverter.java

```diff
@@ -220,21 +233,44 @@ protected Map<String, String> configureConverterType(boolean isKey, Map<String,
 
     @Override
     public byte[] fromConnectData(String topic, Schema schema, Object value) {
+        return this.fromConnectData(topic, null, schema, value);
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, Headers headers, Schema schema, Object value) {
         if (schema == null || value == null) {
             return null;
         }
-        if (!Envelope.isEnvelopeSchema(schema)) {
-            // TODO Handling of non-data messages like schema change or transaction metadata
-            return null;
+
+        if (this.sourceLocation == SourceLocation.VALUE) {
+            if (!Envelope.isEnvelopeSchema(schema)) {
+                // TODO Handling of non-data messages like schema change or transaction metadata
+                return null;
+            }
+        }
+        else {
+            if (headers.lastHeader(Envelope.FieldName.SOURCE) == null || headers.lastHeader(Envelope.FieldName.OPERATION) == null
+                    || headers.lastHeader(Envelope.FieldName.TRANSACTION) == null) {
```

Comment:
`transaction` metadata are optional and require transactio metadata to be enabled. Is it intentional that it is mandatory here?

Reply:
removed that check